File: app/src/main/java/com/digicoffer/lauditor/Dashboard/DahboardModels/CardDataModels.java
package com.digicoffer.lauditor.Dashboard.DahboardModels;

import java.util.List;
import java.util.Map;

public final class CardDataModels {

    private CardDataModels() {}

    public static class BadgeData {
        public double count;
        public String label;
        public String color;
        public BadgeData(double count, String label, String color) {
            this.count = count; this.label = label; this.color = color;
        }
    }

    // ── Today's Activities ────────────────────────────────────────────────

    public static class MeetingCardData {
        public boolean hasMeeting;
        public int     meetingsCount;
        public String  subject, date, startTime, endTime, meetingType;
        public BadgeData badge;
    }

    public static class AppointmentCardData {
        public boolean hasAppointment;
        public String  subject, date, startTime, endTime;
        public BadgeData badge;
    }

    public static class MessagesCardData {
        public boolean hasMessages;
        public int     clientMessages, teamMessages;
        public BadgeData badge;
    }

    public static class NotificationCardData {
        public boolean hasNotifications;
        public int     unreadCount;
        public BadgeData badge;
        public List<NotificationItem> notifications;

        public static class NotificationItem {
            public String  id, message, time, date;
            public boolean isRead;
        }
    }

    // ── Analytics Overview ────────────────────────────────────────────────

    public static class TrendCardData {
        public String cardKey;       // "revenueTrend" | "appointmentTrend"
        public String defaultRange;
        public String currencySymbol;
        public Map<String, TrendRange> ranges;
        public BadgeData badge;

        public static class TrendRange {
            public List<String>  xLabels;
            public String        xKey;
            public double        yMin, yMax, yStep;
            public String        yUnit, yLabel;
            public List<DataSet> datasets;
        }

        public static class DataSet {
            public String       name, color;
            public List<Double> data;
        }
    }

    // ─── Add / update these fields inside CardDataModels.MatterCardData ──────────
//
// API JSON shape:
// {
//   "grandTotal": 635,
//   "activeTotalCount": 615,
//   "closedTotalCount": 20,
//   "chartData": {
//     "x_axis": { "key": "count", "min": 0, "max": 540, "step": 135 },
//     "y_axis": { "key": "type",  "labels": ["Legal", "General"] },
//     "datasets": [
//       { "name": "Active",  "color": "blue",   "data": [527, 88] },
//       { "name": "Closed",  "color": "purple",  "data": [11,  9] }
//     ]
//   }
// }
// ─────────────────────────────────────────────────────────────────────────────

// ── Replace MatterCardData in CardDataModels.java ────────────────────────────
// Added: public BadgeData badge;  (parser sets this via parseBadge())
// ─────────────────────────────────────────────────────────────────────────────

    public static class MatterCardData {
        public BadgeData badge;            // ← ADD THIS — parser sets m.badge
        public boolean   hasMatters;
        public int       grandTotal;       // 635 — shown in blue circle badge
        public int       activeTotalCount; // 615 — legend count for dataset[0]
        public int       closedTotalCount; // 20  — legend count for dataset[1]
        public ChartData chartData;

        public static class ChartData {
            public XAxisConfig  xAxis;     // from x_axis { min, max, step }
            public List<String> yLabels;   // from y_axis.labels ["Legal","General"]
            public List<DataSet> datasets;
        }

        public static class XAxisConfig {
            public double min;             // 0
            public double max;             // 540
            public double step;            // 135
        }

        public static class DataSet {
            public String      name;       // "Active" / "Closed"
            public String      color;      // "blue"   / "purple"
            public List<Float> data;       // [527, 88]
        }
    }

// ─── Gson / JSON parsing note ──────────────────────────────────────────────
// If you use Gson, add @SerializedName annotations:
//
//   @SerializedName("x_axis")  public XAxisConfig xAxis;
//   @SerializedName("labels")  public List<String> yLabels;  // nested inside y_axis
//
// Or flatten manually in your API response mapper:
//   cardData.chartData.yLabels = response.chartData.yAxis.labels;
//   cardData.chartData.xAxis   = response.chartData.xAxis;

    public static class StorageCardData {
        public double usedGB, availableGB, totalGB, usagePercentage;
        public String status, displayText;
        public BadgeData badge;
    }

    // ── Business Metrics ──────────────────────────────────────────────────
    // Parsed from a single "hours" API card but shown as TWO separate cards

    /** Card: Total Billable Hours + Non-Billable Hours with donut chart */
    public static class BillableCardData {
        public String billableHours;
        public double billablePercentage;
        public String nonBillableHours;
        public double nonBillablePercentage;
        public String dateRangeLabel;
        public BadgeData badge;
    }

    /** Card: Approximate Revenue + Average Billing Rate */
    public static class ApproxRevenueCardData {
        public BadgeData badge;
        public double approxRevenue;
        public double averageBillingRate;
        public String dateRangeLabel;
        public String currencyCode;
        public String currencySymbol;

        // Add these new fields for the approximate_revenue card
        public int appointmentCount;
        public String description;
    }

    public static class SubscriptionCardData {
        public boolean isPaid, isActive, isExpired, canUpgrade, canPayNow;
        public String  model, planLabel, validityText, validTill;
        public CtaData cta;
        public  StatusBadge status;
        public  static class StatusBadge{
            public String label,color;
        }

        public static class CtaData {
            public String label, style, action;
        }
    }

    // ── Extras ────────────────────────────────────────────────────────────

    public static class HiringCardData {
        public int totalMembers;
        public List<GroupItem> groups;
        public BadgeData badge;
        public static class GroupItem {
            public String id, name;
            public int    memberCount;
        }
    }
}

File: app/src/main/java/com/digicoffer/lauditor/Dashboard/DahboardModels/CardDataModels.kt
package com.digicoffer.lauditor.Dashboard.DahboardModels

class CardDataModels private constructor() {

    class BadgeData(
        @JvmField var count: Double,
        @JvmField var label: String?,
        @JvmField var color: String?
    )

    // ── Today's Activities ────────────────────────────────────────────────

    class MeetingCardData {
        @JvmField var hasMeeting: Boolean = false
        @JvmField var meetingsCount: Int = 0
        @JvmField var subject: String? = null
        @JvmField var date: String? = null
        @JvmField var startTime: String? = null
        @JvmField var endTime: String? = null
        @JvmField var meetingType: String? = null
        @JvmField var badge: BadgeData? = null
    }

    class AppointmentCardData {
        @JvmField var hasAppointment: Boolean = false
        @JvmField var subject: String? = null
        @JvmField var date: String? = null
        @JvmField var startTime: String? = null
        @JvmField var endTime: String? = null
        @JvmField var badge: BadgeData? = null
    }

    class MessagesCardData {
        @JvmField var hasMessages: Boolean = false
        @JvmField var clientMessages: Int = 0
        @JvmField var teamMessages: Int = 0
        @JvmField var badge: BadgeData? = null
    }

    class NotificationCardData {
        @JvmField var hasNotifications: Boolean = false
        @JvmField var unreadCount: Int = 0
        @JvmField var badge: BadgeData? = null
        @JvmField var notifications: List<NotificationItem>? = null

        class NotificationItem {
            @JvmField var id: String? = null
            @JvmField var message: String? = null
            @JvmField var time: String? = null
            @JvmField var date: String? = null
            @JvmField var isRead: Boolean = false
        }
    }

    // ── Analytics Overview ────────────────────────────────────────────────

    class TrendCardData {
        @JvmField var cardKey: String? = null // "revenueTrend" | "appointmentTrend"
        @JvmField var defaultRange: String? = null
        @JvmField var currencySymbol: String? = null
        @JvmField var ranges: Map<String, TrendRange>? = null
        @JvmField var badge: BadgeData? = null

        class TrendRange {
            @JvmField var xLabels: List<String>? = null
            @JvmField var xKey: String? = null
            @JvmField var yMin: Double = 0.0
            @JvmField var yMax: Double = 0.0
            @JvmField var yStep: Double = 0.0
            @JvmField var yUnit: String? = null
            @JvmField var yLabel: String? = null
            @JvmField var datasets: List<DataSet>? = null
        }

        class DataSet {
            @JvmField var name: String? = null
            @JvmField var color: String? = null
            @JvmField var data: List<Double>? = null
        }
    }

    class MatterCardData {
        @JvmField var badge: BadgeData? = null
        @JvmField var hasMatters: Boolean = false
        @JvmField var grandTotal: Int = 0
        @JvmField var activeTotalCount: Int = 0
        @JvmField var closedTotalCount: Int = 0
        @JvmField var chartData: ChartData? = null

        class ChartData {
            @JvmField var xAxis: XAxisConfig? = null
            @JvmField var yLabels: List<String>? = null
            @JvmField var datasets: List<DataSet>? = null
        }

        class XAxisConfig {
            @JvmField var min: Double = 0.0
            @JvmField var max: Double = 0.0
            @JvmField var step: Double = 0.0
        }

        class DataSet {
            @JvmField var name: String? = null
            @JvmField var color: String? = null
            @JvmField var data: List<Float>? = null
        }
    }

    class StorageCardData {
        @JvmField var usedGB: Double = 0.0
        @JvmField var availableGB: Double = 0.0
        @JvmField var totalGB: Double = 0.0
        @JvmField var usagePercentage: Double = 0.0
        @JvmField var status: String? = null
        @JvmField var displayText: String? = null
        @JvmField var badge: BadgeData? = null
    }

    class BillableCardData {
        @JvmField var billableHours: String? = null
        @JvmField var billablePercentage: Double = 0.0
        @JvmField var nonBillableHours: String? = null
        @JvmField var nonBillablePercentage: Double = 0.0
        @JvmField var dateRangeLabel: String? = null
        @JvmField var badge: BadgeData? = null
    }

    class ApproxRevenueCardData {
        @JvmField var badge: BadgeData? = null
        @JvmField var approxRevenue: Double = 0.0
        @JvmField var averageBillingRate: Double = 0.0
        @JvmField var dateRangeLabel: String? = null
        @JvmField var currencyCode: String? = null
        @JvmField var currencySymbol: String? = null
        @JvmField var appointmentCount: Int = 0
        @JvmField var description: String? = null
    }

    class SubscriptionCardData {
        @JvmField var isPaid: Boolean = false
        @JvmField var isActive: Boolean = false
        @JvmField var isExpired: Boolean = false
        @JvmField var canUpgrade: Boolean = false
        @JvmField var canPayNow: Boolean = false
        @JvmField var model: String? = null
        @JvmField var planLabel: String? = null
        @JvmField var validityText: String? = null
        @JvmField var validTill: String? = null
        @JvmField var cta: CtaData? = null
        @JvmField var status: StatusBadge? = null

        class StatusBadge {
            @JvmField var label: String? = null
            @JvmField var color: String? = null
        }

        class CtaData {
            @JvmField var label: String? = null
            @JvmField var style: String? = null
            @JvmField var action: String? = null
        }
    }

    class HiringCardData {
        @JvmField var totalMembers: Int = 0
        @JvmField var groups: List<GroupItem>? = null
        @JvmField var badge: BadgeData? = null

        class GroupItem {
            @JvmField var id: String? = null
            @JvmField var name: String? = null
            @JvmField var memberCount: Int = 0
        }
    }
}
